llvm/test/CodeGen/AMDGPU/GlobalISel/inst-select-load-flat.mir

```diff
@@ -59,7 +59,7 @@ body: |
     ; GFX12-NEXT: [[COPY:%[0-9]+]]:vreg_64 = COPY $vgpr0_vgpr1
     ; GFX12-NEXT: [[FLAT_LOAD_DWORD:%[0-9]+]]:vgpr_32 = FLAT_LOAD_DWORD [[COPY]], 0, 0, implicit $exec, implicit $flat_scr :: (load (s32))
     ; GFX12-NEXT: $vgpr0 = COPY [[FLAT_LOAD_DWORD]]
-    %0:vgpr(p1) = COPY $vgpr0_vgpr1
+    %0:vgpr(p0) = COPY $vgpr0_vgpr1
     %1:vgpr(s32) = G_LOAD %0 :: (load (s32), align 4, addrspace 0)
```

Comment:
surprised to see that even with `p1`, it can generate `FLAT_LOAD_DWORD`

Reply:
The patterns don't know the pointer type, only the memory operand address space